narwhals/_spark_like/dataframe.py

```diff
@@ -274,6 +275,14 @@ def select(
     ) -> Self:
         new_columns = evaluate_exprs(self, *exprs)
         new_columns_list = [col.alias(col_name) for (col_name, col) in new_columns]
+        if not self._implementation.is_sqlframe():
+            from pyspark.errors import AnalysisException
+
+            try:
+                return self._with_native(self.native.select(*new_columns_list))
+            except AnalysisException as e:
+                msg = f"Selected columns not found in the DataFrame.\n\nHint: Did you mean one of these columns: {self.columns}?"
```

Comment:
Not 100% sure about this error message. I don't we can access the missing column names at this level, am I missing something?

Reply:
I think what you're written is great - even though we can't access them, we can still try to be helpful